File: problem=057/Solution.java
import java.math.BigInteger;

final class Solution {

	private static final BigInteger TWO = new BigInteger("2");
	private static final BigInteger THREE = new BigInteger("3");

	private static int solve() {
		BigInteger n = BigInteger.ONE, d = TWO;

		int answer = 0;
		for (int i = 0 ; i < 1000; i++) {
			BigInteger nn = d, nd = d.multiply(TWO).add(n), zn = d.multiply(THREE).add(n);

			if (zn.toString().length() > nd.toString().length()) {
				answer++;
			}

			n = nn;
			d = nd;
		}
		return answer;
	}

	public static void main(String[] args) throws Exception {
		System.out.println(solve());
	}
}

// If this solution helped you, please star the repo!

File: problem=057/Solution.js
const solve = () => {
	let arr = [1, 2];
	for (let i = 0; i < 10; i++) {
		const [n, d] = arr;
		const fn = [d, 2 * d + n];
		const z = [3 * d + n, 2 * d + n];
		console.log(z);
		arr = fn;
	}
};

console.log(solve());
